lan867x/src/esp_eth_phy_lan867x.c

```diff
@@ -0,0 +1,319 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <string.h>
+#include <stdlib.h>
+#include <sys/cdefs.h>
+#include "esp_log.h"
+#include "esp_check.h"
+#include "esp_eth_phy_802_3.h"
+#include "esp_eth_driver.h"
+#include "esp_eth_phy_lan867x.h"
+
+static const char *TAG = "lan867x";
+
+/***************List of Supported Models***************/
+#define LAN867X_MODEL_NUM  0x16
+#define LAN867X_OUI 0xC0001C
+
+static const uint8_t supported_models[] = {
+    LAN867X_MODEL_NUM,
+};
+
+/***************Vendor Specific Register***************/
+typedef union {
+    struct {
+        uint32_t oui_bits_2_9 : 8; /*!< Organizationally Unique Identifier(OUI) bits 3 to 10 */
+        uint32_t oui_bits_10_17 : 8; /*!< Organizationally Unique Identifier(OUI) bits 11 to 18 */
+    };
+    uint32_t val;
+} lan867x_phyidr1_reg_t;
+#define ETH_PHY_IDR1_REG_ADDR (0x02)
+
+typedef union {
+    struct {
+        uint32_t model_revision : 4; /*!< Model revision number */
+        uint32_t vendor_model : 6;   /*!< Vendor model number */
+        uint32_t oui_bits_18_23 : 6; /*!< Organizationally Unique Identifier(OUI) bits 19 to 24 */
+    };
+    uint32_t val;
+} lan867x_phyidr2_reg_t;
+#define ETH_PHY_IDR2_REG_ADDR (0x03)
+
+typedef union {
+    struct {
+        uint32_t reserved1 : 14;    // Reserved
+        uint32_t rst : 1;           // PLCA Reset
+        uint32_t en : 1;            // PLCA Enable
+    };
+    uint16_t val;
```

Comment:
Should it be `uint32_t`?

Reply:
I see, `esp_eth_phy_802_3_read/write_mmd_register` accept `uint16_t`. We need to change it to `uint32_t` to all register definitions are the same. Otherwise it would be kind of messy... Please fix here and it ESP-IDF.

Reply:
But MDIO registers always 16 bit wide, aren't they ? To me it makes sense to use `uint16_t` wherever they are used.

Reply:
Bit fields can't have size of 16 bit on 32-bit architecture (https://en.cppreference.com/w/c/language/bit_field). You can see it even in your code (you use `uint32_t`). So the size of the datatype is 4 bytes anyway. In addition, your code could have some troubles with portability, I guess.

Reply:
Done.